src/components/List/index.js

```diff
@@ -29,6 +30,10 @@ export default function List (props) {
   const filteredDataByCategory = category === 'All' ? filteredData : filteredData.filter(listing => listing.category === category);
   filteredDataByCategory.sort((a, b) => a.timestamp > b.timestamp ? 1 : -1);
 
+  if (!filteredDataByCategory.length) {
```

Comment:
You might want to set the default state of the `data` to be `null` and have your `isLoading` logic based off that instead. It seems in this case that if a category has no results it will appear to be loading.

Reply:
Thanks for catching that!